libs/math/tests/unit/fixed_point/fixed_point.cpp

```diff
@@ -236,6 +236,129 @@ TEST(FixedPointTest, Conversion_64_64)
+  EXPECT_EQ(large2, fp32_t::FP_MIN);
+
+  fp32_t e3("0E-12fp32");
+  EXPECT_EQ(e3, fp32_t::_0);
+}
+
+TEST(FixedPointTest, FromString_32_32)
+{
+  // Get raw value
+  fp64_t one("1fp64_t");
+  fp64_t zero_point_five("0.5fp64_t");
+  fp64_t one_point_five("1.5fp64_t");
+  fp64_t two_point_five("2.5fp64_t");
+  fp64_t m_one_point_five("-1.5fp64_t");
+  fp64_t m_one_point_five_em10("-1.5e-5fp64_t");
+  fp64_t m_one_point_five_e10("-1.5e+5fp64_t");
+  fp64_t m_one_point_five_flt(-1.5);
+
+  EXPECT_EQ(zero_point_five.Data(), 0x080000000);
+  EXPECT_EQ(one.Data(), 0x100000000);
+  EXPECT_EQ(one_point_five.Data(), 0x180000000);
+  EXPECT_EQ(two_point_five.Data(), 0x280000000);
+  EXPECT_EQ(m_one_point_five, m_one_point_five_flt);
+  fp64_t m_fifteen_million(-150000);
+  fp64_t m_tiny(-0.000015);
+
+  EXPECT_EQ(m_one_point_five_em10, m_tiny);
+  EXPECT_EQ(m_one_point_five_e10, m_fifteen_million);
+
+  fp64_t e1("2.718281828459045235360287471352662498");
+  fp64_t e2(2.718281828459045235360287471352662498);
+  EXPECT_TRUE(e1.Near(e2));
+  EXPECT_NE(e1, e2);
+
+  fp64_t::StateClear();
+  fp64_t large1("1442695040888963407359924681001892137");
+  EXPECT_TRUE(fp64_t::IsStateOverflow());
+  fp64_t::StateClear();
+  fp64_t large2("-1442695040888963407359924681001892137");
+  EXPECT_TRUE(fp64_t::IsStateOverflow());
+  EXPECT_EQ(large1, fp64_t::FP_MAX);
+  EXPECT_EQ(large2, fp64_t::FP_MIN);
+
+  fp64_t e3("0E-12fp64");
+  EXPECT_EQ(e3, fp64_t::_0);
+}
+
+TEST(FixedPointTest, FromString_64_64)
+{
+  // Get raw value
+  fp128_t one("1fp128");
+  fp128_t zero_point_five("0.5fp128");
+  fp128_t one_point_five("1.5fp128");
+  fp128_t two_point_five("2.5fp128");
+  fp128_t m_one_point_five("-1.5fp128");
+  fp128_t m_one_point_five_em10("-1.5e-10fp128");
+  fp128_t m_one_point_five_e10("-1.5e+10fp128");
+  fp128_t m_one_point_five_flt(-1.5);
+
+  EXPECT_EQ(zero_point_five.Data(), static_cast<int128_t>(0x8000000000000000));
+  EXPECT_EQ(one.Data(), static_cast<int128_t>(1) << 64);
+  EXPECT_EQ(one_point_five.Data(), static_cast<int128_t>(0x18) << 60);
+  EXPECT_EQ(two_point_five.Data(), static_cast<int128_t>(0x28) << 60);
+  EXPECT_EQ(m_one_point_five, m_one_point_five_flt);
+  fp128_t m_fifteen_million(-15000000000);
+  fp128_t m_tiny(-0.00000000015);
+
+  EXPECT_EQ(m_one_point_five_em10, m_tiny);
+  EXPECT_EQ(m_one_point_five_e10, m_fifteen_million);
+
+  fp128_t e1("2.718281828459045235360287471352662498");
+  fp128_t e2(2.718281828459045235360287471352662498);
+  EXPECT_TRUE(e1.Near(e2));
+  EXPECT_NE(e1, e2);
+
+  fp128_t::StateClear();
+  fp128_t large1("1442695040888963407359924681001892137");
+  EXPECT_TRUE(fp128_t::IsStateOverflow());
+  fp128_t::StateClear();
+  fp128_t large2("-1442695040888963407359924681001892137");
+  EXPECT_TRUE(fp128_t::IsStateOverflow());
+  EXPECT_EQ(large1, fp128_t::FP_MAX);
+  EXPECT_EQ(large2, fp128_t::FP_MIN);
+
+  fp128_t e3("0E-12fp128");
+  EXPECT_EQ(e3, fp128_t::_0);
+}
+
 TEST(FixedPointTest, Constants_16_16)
```

Comment:
can you please add a few tests to check various parsing cases; e.g.:
- FixedPoint("000000000000001.0000000000000000000000000000000000000000000000001")
- FixedPoint("000000000000001.0000000000000000000000000000000000000000000000000")
etc.

Reply:
ok

Reply:
resolved.